File: app/src/main/java/com/worldventures/dreamtrips/wallet/analytics/oncard/action/SmartCardUserAction.java
package com.worldventures.dreamtrips.wallet.analytics.oncard.action;

import io.techery.janet.smartcard.model.analytics.AnalyticsLog;

public abstract class SmartCardUserAction extends SmartCardAnalyticsAction {

   SmartCardUserAction(AnalyticsLog logEntry) {
      super(logEntry);
   }

   public abstract void setUserId(int userId);

   @Override
   protected void processLog(int type, AnalyticsLog logEntry) {
      super.processLog(type, logEntry);
      switch (type) {
         case AnalyticsLog.USER_ASSIGNED:
            attributeMap.put("ocassigned", "1");
            break;
         case AnalyticsLog.USER_UNASSIGNED:
            attributeMap.put("ocunassigned ", "1");
            break;
      }
   }

}

File: wallet/src/main/java/com/worldventures/wallet/analytics/oncard/action/SmartCardUserAction.kt
package com.worldventures.wallet.analytics.oncard.action

import io.techery.janet.smartcard.model.analytics.AnalyticsLog

abstract class SmartCardUserAction internal constructor(logEntry: AnalyticsLog) : SmartCardAnalyticsAction(logEntry) {

   abstract fun setUserId(userId: Int)

   override fun processLog(type: Int, logEntry: AnalyticsLog) {
      super.processLog(type, logEntry)
      when (type) {
         AnalyticsLog.USER_ASSIGNED -> attributeMap.put("ocassigned", "1")
         AnalyticsLog.USER_UNASSIGNED -> attributeMap.put("ocunassigned ", "1")
      }
   }

}